qemu/tests/block_copy.py

```diff
@@ -264,9 +264,21 @@ def get_image_file(self):
         blocks = self.vm.monitor.info("block")
         try:
             if isinstance(blocks, str):
+                # ide0-hd0: removable=1 locked=0 file=/tmp/test.img
                 image_regex = '%s.*\s+file=(\S*)' % self.device
                 image_file = re.findall(image_regex, blocks)
-                return image_file[0]
+                if image_file:
+                    return image_file[0]
+                # ide0-hd0 (#block184): a b c
+                # or
+                # ide0-hd0 (#block184): a b c (raw)
+                image_file = re.findall("%s[^:]+: ([^(]+)\(?" % self.device,
+                                        blocks)
+                if image_file:
+                    if image_file[0][-1] == ' ':
+                        return image_file[0][:-1]
+                    else:
+                        return image_file[0]
```

Comment:
This fixes errors such as:

```
Reproduced traceback from: /home/cleber/src/avocado/avocado-vt/avocado_vt/test.py:407
Traceback (most recent call last):
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_commit_stress.py", line 89, in run
    stress_test = BlockCommitStress(test, params, env, tag)
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_copy.py", line 48, in __init__
    self.image_file = self.get_image_file()
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_copy.py", line 268, in get_image_file
    return image_file[0]
IndexError: list index out of range
```

When running a test such as `type_specific.io-github-autotest-qemu.block_commit.with_payload.with_stress`.

ACK.